ui/react-components/components/GridSummary/GridSummary.jsx

```diff
@@ -0,0 +1,107 @@
+import React from "react";
+import { injectIntl } from "react-intl";
+import "./GridSummary.module.scss";
+import moment from "moment";
+import { sortBy } from "lodash";
+import classNames from 'classnames'
+
+const transformGridData=(gridData)=>{
+  return gridData.map(dataElement => {
+    let rowList = [];
+    for (let element in dataElement.appointmentCountMap) {
+      rowList.push({
+        date: element,
+        count: dataElement.appointmentCountMap[element].allAppointmentsCount,
+        missedCount:
+          dataElement.appointmentCountMap[element].missedAppointmentsCount
+      });
+    }
+    return {
+      rowLabel: dataElement.appointmentService.name,
+      rowDataList: sortBy(rowList, dateObj => moment(dateObj.date))
+    };
+  });
+}
+
+const GridSummary = props => {
+  const { gridData=[], weekStartDate = moment().startOf("isoweek"), onClick } = props;
```

Comment:
can we pass weekStartDate as props from parent component instead of hardcoding here??

Reply:
Yes. In current code , we can get weekStartDate from props as well. We just set the default value of weekStartDate to current week start date in component